sudokuGUI.py

```diff
@@ -1,532 +1,533 @@
+                                            text = "Hard" ))
+        
+        self.menu_buttons.append(Button(200, 460, 200, 80,
+                                            function  = self.get_board,
+                                            params = 3,
+                                            colour = WHITE,
+                                            text = "Extreme" ))
+
+
+        # PLAYING BUTTONS
+        self.playing_buttons.append(Button(20, 40, 100, 40,
+                                            function = self.solved_gui,
+                                            text = "Solve" ))
+        
+        self.playing_buttons.append(Button(140, 40, 100, 40, 
+                                            function  = self.hint,
+                                            text = "Hint" ))
+        
+        self.playing_buttons.append(Button(260, 40, 100, 40, 
+                                            function = self.try_again, 
+                                            highlighted_colour = PENCIL_GRAY,
+                                            text = "Menu"))
+
+        # END GAME BUTTONS
+        self.end_game_buttons.append(Button(150, 350, 120, 60,
+                                            function = self.exit_game,
+                                            colour = WHITE,
+                                            highlighted_colour = PENCIL_GRAY,
+                                            text = "Exit" ))
+        
+        self.end_game_buttons.append(Button(300, 350, 120, 60, 
+                                            function  = self.try_again,
+                                            colour = WHITE,
+                                            highlighted_colour = PENCIL_GRAY,
+                                            text = "Try Again" ))
+
+
+    # Function that puts the numbers on the screen
+    # Input: Window element, text string and position to put the text in and option (1 - correct ans, 2 - pencil option)
+    # Output: Nothing
+    def text_to_screen(self, window, text : str, pos : list, option : int):
+        font = self.font.render(text, False, BLACK if option == 1 else PENCIL_GRAY)
+        font_height = font.get_height()
+        font_width = font.get_width()
+        pos[0] += (CELL_SIZE - font_width) // 2
+        pos[1] += (CELL_SIZE - font_height) // 2
+        window.blit(font, pos)
+
+
+    # Function that shades the permenent cells on the board
+    # Input: The window element and a list of cells to lock
+    # Output: Nothing.
+    def shade_locked_cells(self, window, lock_cells : list):
+        for cell in lock_cells:
+            pygame.draw.rect(window, GRAY, (cell[0] * CELL_SIZE + X_GRID, cell[1] * CELL_SIZE + Y_GRID, CELL_SIZE, CELL_SIZE))
+
+
+    # Function that checks if the number is an integer
+    # Input: string
+    # Output: True if the number is a string, False otherwise
+    def is_int(self, string : str) -> bool: 
```

Comment:
There are better ways of doing it which are built in python like `isnumeric()` :)